### PR TITLE
[Bugfix:Edit User] Fix the edit user/new user page

### LINKED ISSUE #4061: "Edit User" panel (from "Manage Students") does not save
**Describe the bug**
Seems like we've broken the "Edit User" panel.  I press "Submit" but nothing seems to save.
It also does not work to create a "New Student" user.

The related panel for "Edit User" from "Manage Graders" and (edit or "New Grader") does seem to work.

Sadly there is no indication of failure.  No info in site_errors.  No red alert box with an error message.  I just don't get the green alert success message and the data doesn't change.

I do get this message when I try to refresh the page (which is unhelpful)

Screen Shot 2019-07-11 at 11 55 31 PM

### PR DESCRIPTION
**Pull request by jmcknight98**
Closes #4061 
### Please check if the PR fulfills these requirements:

* [x] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)

Oops, I broke it (see issue above)... 
Should fix the edit user and New Student page

### CODE REVIEW

File: site/app/templates/admin/users/UserForm.twig [hunk 1]
```diff
@@ -109,7 +109,7 @@
     </script>
 {% endblock %}
 {% block form %}
-    <form method="post" action="{{ return_url_update_grader }}" onchange="userFormChange()">
+    <form method="post" action="{{ core.buildUrl({'component': 'admin', 'page': 'users', 'action': action}) }}" onchange="userFormChange()">
```
Comment (xprilion) [inline site/app/templates/admin/users/UserForm.twig]: I guess the `return_url_update_grader` wasn't being passed to this twig file at all. You might want to shift it in the View file

File: site/app/views/admin/UsersView.php [hunk 2]
```diff
@@ -61,7 +61,6 @@ public function listGraders($graders, $reg_sections, $rot_sections, $use_databas
             "reg_sections" => $reg_sections,
             "rot_sections" => $rot_sections,
             "use_database" => $use_database,
-            "return_url_update_grader" => $this->core->buildUrl(array('component' => 'admin', 'page' => 'users','action' => 'update_grader')),
```
Comment (xprilion) [inline site/app/views/admin/UsersView.php]: Shifting this to `userForm` should help!

Comment (xprilion) [reply site/app/views/admin/UsersView.php]: Might need some modification though

**Comment by jmcknight98** [comment]
Yes it wasn't being passed to the twig file.  I was cleaning up a few of the core.buildUrl calls in twig files (as suggested in #2273) that I happened to notice while working on a separate issue.  But this one was in a file that wasn't included, so I just reverted the change to fix the bug.  It can be cleaned up as part of that issue.

**Comment by zjxiaohan** [comment]
Sorry I just saw your comment after sending the comment.

Yes we could have a quick fix temporarily and I will do a cleanup some time after, but note that this fix is incomplete as it breaks edit grader function and needs to be fixed.

A quick fix would be changing line 55 in `GraderList.twig` to use `with {'action': 'update_grader}`

**Comment by jmcknight98** [comment]
Oh yes I didn't notice they were using the same form when doing the quick fix, they should both work now as intended.

File: site/app/views/admin/UsersView.php (hunk 2 above)
Comment (jmcknight98) [reply site/app/views/admin/UsersView.php]: I switched it back to using buildUrl in the twig file, its more complicated then I thought so trying to fix it as I passed through the files was a mistake.  This should be changed in the future within #2273, but for now changes are just to make it work as before.

**Pull request merged**